edx_sga/static/js/src/edx_sga.js

```diff
@@ -292,6 +302,32 @@ function StaffGradedAssignmentXBlock(runtime, element) {
                     });
                 block.find('#staff-debug-info-button')
                     .leanModal();
+
+                $(element).find('#download-init-button').click(function(e) {
+                  e.preventDefault();
+                  var self = this;
+                  $(self).addClass("disabled");
+                  $.get(
+                    prepareDownloadSubmissionsUrl,
+                    function(data) {
+                      if (data["downloadable"]) {
+                        window.location = downloadSubmissionsUrl;
+                        $(element).find('.task-message').html("");
+                      } else {
+                        $(element).find('.task-message').html(
+                          interpolate(
+                            gettext(
+                              'Started preparing student submissions zip file. Please come back latter and click %(button)s again.'
+                            ),
+                            {button: '<b>Download All Submission</b>'},
```

Comment:
2 things:

1. ~~latter~~ -> later
1. The text for the "Download All Submissions" button is translated. I think we should just add "Download All Submissions" directly into the string and remove the call to `interpolate`.
1. In a previous version of this PR, you introduced a polling request that checked for the zip file every N seconds. Is that not possible to do here  instead of telling the user to "come back later"? You could show some "Preparing..." message, make the request to compress the files, then start the polling function. When the polling function got a response that said the file was ready, you could update the UI to remove that "Preparing..." message.

Reply:
on refreshing page is preparing message is remove